library/network/test/cwm_firewall_interfaces_test.rb

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env rspec
+
+require_relative "test_helper"
+
+Yast.import "CWMFirewallInterfaces"
+Yast.import "NetworkInterfaces"
+Yast.import "Mode"
+Yast.import "UI"
+
+describe Yast::CWMFirewallInterfaces do
+
+  subject { Yast::CWMFirewallInterfaces }
+
+  describe "#CreateOpenFirewallWidget" do
+    let(:widget_settings) { { "services" => ["apache"] } }
+  end
+
+  describe "#OpenFirewallInit" do
+    let(:open_firewall_widget) { false }
+    let(:widget_settings) { { "services" => ["apache"] } }
+    let(:allowed_interfaces) { ["eth0"] }
+    let(:all_interfaces) { ["eth0", "eth1"] }
+    let(:firewalld_enabled?) { true }
+
+    before do
+      allow(Yast::UI).to receive(:WidgetExists).with(Id("_cwm_open_firewall"))
+        .and_return(open_firewall_widget)
+      allow(subject).to receive(:InitAllInterfacesList)
+      allow(subject).to receive(:InitAllowedInterfaces)
+      allow(subject).to receive(:UpdateFirewallStatus)
+      allow(subject).to receive(:EnableOrDisableFirewallDetails)
+      allow(subject).to receive(:allowed_interfaces).and_return(allowed_interfaces)
+      allow(subject).to receive(:all_interfaces).and_return(all_interfaces)
+      allow(Yast::UI).to receive(:ChangeWidget)
+      allow_any_instance_of(Y2Firewall::Firewalld)
+        .to receive(:enabled?).and_return(firewalld_enabled?)
+    end
+
+    context "when the open firewall widget does not exist" do
+      it "return nil" do
+        subject.OpenFirewallInit(widget_settings, "")
+      end
+    end
+
+    context "when the open firewall widget exist" do
+      let(:open_firewall_widget) { true }
+
+      it "initializes the list of network interfaces" do
+        expect(subject).to receive(:InitAllInterfacesList)
+        subject.OpenFirewallInit(widget_settings, "")
+      end
+
+      it "initializes the list of allowed interfaces" do
+        expect(subject).to receive(:InitAllowedInterfaces)
+        subject.OpenFirewallInit(widget_settings, "")
+      end
```

Comment:
double checking? the same test twice ...

Reply:
It's defined twice? one is for all the interfaces and the other for the allowed ones.

Reply:
ah yes. sorry